packages/tools/devtools/devtools-test-app/src/FluidObject.ts

```diff
@@ -179,39 +179,56 @@ export class AppData extends DataObject {
 		const builder = new SchemaFactory("DefaultVisualizer_SharedTree_Test");
 
 		// TODO: Maybe include example handle
+		class LeafNodeSchema extends builder.object("leaf-node", {
+			value: [builder.boolean, builder.handle, builder.string],
+		}) {}
+
+		class ArrayNodeSchema extends builder.object("array-node", {
+			value: [builder.string, builder.boolean],
+			childLeaf: builder.optional(LeafNodeSchema),
+		}) {}
+
+		class BranchNodeSchema extends builder.object("branch-node", {
+			arrayOrHandleNode: [builder.array(ArrayNodeSchema), builder.handle],
+			numericLeafNode: builder.number,
+		}) {}
 
-		class LeafSchema extends builder.object("leaf-item", {
-			leafField: [builder.boolean, builder.handle, builder.string],
+		class MainBranchA extends builder.object("main-object-a", {
+			numericLeafNode: builder.number,
+			objectNode: BranchNodeSchema,
 		}) {}
 
-		class ChildSchema extends builder.object("child-item", {
-			childField: [builder.string, builder.boolean],
-			childData: builder.optional(LeafSchema),
+		class MainBranchB extends builder.object("main-object-b", {
+			numericLeafNode: builder.number,
+			booleanValue: builder.boolean,
 		}) {}
 
-		class RootNodeSchema extends builder.object("root-item", {
-			childrenOne: builder.array(ChildSchema),
-			childrenTwo: builder.number,
+		class RootField extends builder.object("root-field-b", {
```

Comment:
Nit: this naming is a bit confusing. This schema we've defined is a node schema, but we've called it `RootField`.